codegens/golang/lib/index.js

```diff
@@ -243,7 +244,11 @@ self = module.exports = {
     }
     codeSnippet += `${indent}"net/http"\n${indent}"io/ioutil"\n)\n\n`;
 
-    codeSnippet += `func main() {\n\n${indent}url := "${encodeURI(request.url.toString())}"\n`;
+    finalUrl = new sdk.Url(request.url.toString());
+    // URL encoding each part of Url individually
+    finalUrl = `${finalUrl.protocol}://${finalUrl.getRemote()}${finalUrl.getPathWithQuery(true)}`;
```

Comment:
A url may not have protocol present. For example if the url is `postman.com/post`, then `finalURL.protocol` will return `undefined` and the finalUrl will be `undefined://postman.com/post` which is invalid. 

Infact I don't see why we need to manually construct the final URL. [url.toString()](https://github.com/postmanlabs/postman-collection/blob/cd3fa984918bf5edd6c96a6fbb2886a573439ebd/lib/collection/url.js#L197) does that for us.

Reply:
Ah, sorry. I was unfamiliar with `url.toString()`. I will make a necessary changes as soon as possible.